src/app/hooks/useClickTrackerHandler/index.jsx

```diff
@@ -33,107 +31,186 @@ const useClickTrackerHandler = (props = {}) => {
-            };
-
-            optimizely.track(
-              optimizelyMetricNameOverride
-                ? `${optimizelyMetricNameOverride}_clicks`
-                : 'component_clicks',
-              optimizely.user.id,
-              overrideAttributes,
-            );
-          }
+      if (shouldSendEvent) {
+        const nextPageUrl = event?.currentTarget?.href;
+
+        event.stopPropagation();
+        event.preventDefault();
+
+        if (optimizely) {
+          const eventName = OPTIMIZELY_CONFIG.viewClickAttributeId;
+
+          const overrideAttributes = {
+            ...optimizely.user.attributes,
+            [`clicked_${eventName}`]: true,
+          };
 
-          const optimizelyVariation =
-            optimizely?.getVariation(OPTIMIZELY_CONFIG.ruleKey) || null;
-
-          try {
-            await sendEventBeacon({
-              type: EVENT_TYPE,
-              campaignID,
-              componentName,
-              format,
-              pageIdentifier,
-              platform,
-              producerId,
-              producerName,
-              service,
-              advertiserID,
-              statsDestination,
-              url,
-              detailedPlacement,
-              useReverb,
-              ...(optimizelyVariation &&
-                optimizelyVariation !== 'off' && {
-                  experimentVariant: optimizelyVariation,
-                }),
-            });
-          } finally {
-            if (nextPageUrl && !preventNavigation) {
-              if (optimizely) {
-                optimizely.close();
-              }
-              window.location.assign(nextPageUrl);
+          optimizely.track(
+            optimizelyMetricNameOverride
+              ? `${optimizelyMetricNameOverride}_clicks`
+              : 'component_clicks',
+            optimizely.user.id,
+            overrideAttributes,
+          );
+        }
+
+        const optimizelyVariation =
+          optimizely?.getVariation(OPTIMIZELY_CONFIG.ruleKey) || null;
+
+        try {
+          await sendEventBeacon({
+            ...beaconProps,
+            useReverb,
+            ...(optimizelyVariation &&
+              optimizelyVariation !== 'off' && {
+                experimentVariant: optimizelyVariation,
+              }),
+          });
+        } finally {
+          if (nextPageUrl && !preventNavigation) {
+            if (optimizely) {
+              optimizely.close();
             }
+            window.location.assign(nextPageUrl);
           }
         }
       }
-    },
-    [
-      trackingIsEnabled,
-      clicked,
-      campaignID,
-      componentName,
-      pageIdentifier,
-      platform,
-      preventNavigation,
-      producerId,
-      producerName,
-      service,
-      statsDestination,
-      url,
-      advertiserID,
-      format,
-      optimizely,
-      optimizelyMetricNameOverride,
-      detailedPlacement,
-      useReverb,
-    ],
-  );
+    }
+  }, Object.keys(beaconProps));
```

Comment:
```suggestion
  }, ...beaconProps);
```
Is the change you made here functionally equivalent to what was there before the refactor? I think this is passing the names of the props rather than the props and their values?